### PR TITLE
Compile InfoAtLink.cpp with appropriate flags

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Determine the correct flags for compiling InfoAtLink.cpp and use them in WrapExecutableLinker.sh.

This makes InfoAtLink more robust, enabling #2690 and #2680.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Informer/InfoAtLinkExtractFlags.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("compiler")
```
Comment (nilsdeppe) [inline src/Informer/InfoAtLinkExtractFlags.py:18]: some comments as to why you use argparse and these arguments would be helpful, I think

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Have you tested that things like:
- rerunning CMake without any setting changes doesn't require everything to be relinked
- rebuilding an executable several times without changes results in no link/build

**Comment by nilsvu** [comment]
@nilsdeppe Seems to work ok, at least the scenarios that I tested. Pushed a few fixups to smooth things out. Before, the script re-ran at every build, but executables didn't re-link. Now, re-building with no changes does nothing.

**Review by nilsdeppe — COMMENTED** [review]
LGTM, please go ahead and squash :)

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
@kidder or @wthrowe could you take a look at this one?

File: src/Informer/InfoAtLinkExtractFlags.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+"""
+
+import argparse
+
+parser = argparse.ArgumentParser(description=__doc__)
+# The first positional argument is the compiler
+parser.add_argument("compiler")
+# Strip `InfoAtLink.cpp` file names
+parser.add_argument("-o", required=False)
+parser.add_argument("-c", required=False)
```
Comment (wthrowe) [inline src/Informer/InfoAtLinkExtractFlags.py]: I think the purpose of this code is to remove known options that we don't care about, but -o takes an argument and -c doesn't, so why are these lines the same?

Comment (nilsvu) [reply src/Informer/InfoAtLinkExtractFlags.py]: Thanks for pointing this out. Pushed a fixup to handle this better.

File: src/Informer/InfoAtLinkExtractFlags.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+the compiler.
+"""
+
+import argparse
+
+parser = argparse.ArgumentParser(description=__doc__)
+# The first positional argument is the compiler
+parser.add_argument("compiler")
+# Strip the object file name
+parser.add_argument("-o", required=False)
```
Comment (wthrowe) [inline src/Informer/InfoAtLinkExtractFlags.py]: This should be `required=True` because you use it unconditionally on line 37.

**Review by wthrowe — CHANGES_REQUESTED** [review]
You can squash this directly.

**Comment by wthrowe** [comment]
Looks good.  Squash.

**Review by wthrowe — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**